src/Sprite/OP2BmpLoader.cpp

```diff
@@ -32,6 +31,14 @@ void OP2BmpLoader::ExtractImage(std::size_t index, const std::string& filenameOu
 	BitmapFile::WriteIndexed(filenameOut, imageMeta.GetBitCount(), imageMeta.width, -static_cast<int32_t>(imageMeta.height), palette, pixelContainer);
 }
 
+std::vector<Color> OP2BmpLoader::GetPalette(const ImageMeta& imageMeta)
+{
+	std::vector<Color> palette(artFile.palettes[imageMeta.paletteIndex].size());
+	std::copy(artFile.palettes[imageMeta.paletteIndex].begin(), artFile.palettes[imageMeta.paletteIndex].end(), palette.begin());
+
+	return palette;
+}
```

Comment:
Could this be simplified by using a by-value copy of `std::vector`? Do we want to make copies, or should we use references?

Reply:
I don't have too much of an opinion this time besides keeping it simple. We could assume that ArtFile should not be changing content in this situation and pass a reference if desired.